### DISCUSSION TITLE
I can't deploy lesson 3's code

### DISCUSSION THREAD

**Discussion by chukynya** [smartcontractkit/full-blockchain-solidity-course-py#11] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/77424694/133065142-284791ff-a075-4fd0-bd1e-77585bea7040.png)
![image](https://user-images.githubusercontent.com/77424694/133065210-83f6ba92-fb92-4add-acd7-c7c5dc2395d7.png)

**Comment by PatrickAlphaC** [comment]
Thanks for making this question!

How much Rinkeby ETH do you have in your metamask?

**Reply by chukynya** [reply]
I have 18.74

**Reply by PatrickAlphaC** [reply]
Any chance you could upload a video of what you're doing? It looks like you actually did deploy the contract. You might have to wait a few minutes.

**Reply by chukynya** [reply]
https://user-images.githubusercontent.com/77424694/133186336-2b64a59a-8263-430c-b87e-51296cd9b7fc.mp4

**Reply by PatrickAlphaC** [reply]
oh wow... uh.... hmm....

I really have no idea. Are you able to deploy any other contracts? 

Can you try deleting some functions and see if you can deploy? If you see that error saying "gas estimation failed" that means it won't deploy.

**Reply by chukynya** [reply]
Actually i can deploy any other contract, except a contract with a interface

**Reply by PatrickAlphaC** [reply]
Can you try a different version of solidity maybe? What is your compiler set to?

**Comment by AlbertoSinigaglia** [comment]


**Reply by PatrickAlphaC** [reply]
Did this allow you to deploy lesson 3's code?

**Reply by AlbertoSinigaglia** [reply]
nvm, not relevant to the question, it was just because I was doing some "wrong" stuff in the constructor, but he's not doing it, so this doesn't help, I wrote the possible real answer on the other thread

**Comment by AlbertoSinigaglia** [comment, marked as answer]
The explanation might be pretty simple... and has nothing to do with your contract (infact, if you just use contract test{} you will still have the same error)... it's because in the testnet (at least, idk about the mainnet, i suppose that @PatrickAlphaC might helps here) you can't send transaction while deploying contracts; if you set your value attribute on the left to 0 you will see that it will work fine, but if you set to something more than zero, it will fail

EDIT: TLDR:
Make sure your `value` is set to 0 when deploying.

**Reply by PatrickAlphaC** [reply]
Amazing! You were able to solve this! Great work!

**Reply by chukynya** [reply]
So how can i send transaction ?

**Reply by AlbertoSinigaglia** [reply]
@henrychrist to deploy, set the value to 0,  and then to fund, set the value to whatever you want and press fund

**Reply by chukynya** [reply]
![image](https://user-images.githubusercontent.com/77424694/133704556-1b9f308c-68d0-488a-a1f1-82c0fdfb5291.png)
It's work thank youu
